### PR TITLE
Retire startState

### PR DESCRIPTION
**Pull request by mcostalba**
StateInfo is usually allocated on the stack by search()
and passed in do_move(), this ensures maximum efficiency and
speed and at the same time unlimited move numbers.

The draw back is that to handle Position init we need to
reserve a StateInfo inside Position itself and use at
init time and when copying from another Position.

After lazy SMP we don't need anymore this gimmick and we can
get rid of this special case and always pass an external
StateInfo to Position object.

No functional change.

### CODE REVIEW

File: src/position.h [hunk 1]
```diff
@@ -86,14 +86,13 @@ class Position {
 public:
   static void init();
 
-  Position() = default; // To define the global object RootPos
+  Position() = default;
   Position(const Position&) = delete;
   Position(const Position& pos, Thread* th) { *this = pos; thisThread = th; }
-  Position(const std::string& f, bool c960, Thread* th) { set(f, c960, th); }
-  Position& operator=(const Position&); // To assign RootPos from UCI
```
Comment (zamar) [inline src/position.h]: For consistency

```
Position& operator=(const Position&) = delete;
```

**Comment by zamar** [comment]
approved - one minor comment.

**Comment by mcostalba** [comment]
@zamar Thanks, I am doing additional work, I will incorporate your suggestion on that. When finished I will update the pull request with a new commit for easy reviewing.

**Comment by mcostalba** [comment]
I have submitted a safety test with 3 threads:

```
ELO: -0.00 +-12.7 (95%) LOS: 50.0%
Total: 1000 W: 173 L: 173 D: 654

```

**Comment by zamar** [comment]
@mcostalba: Is the patch ready for final review? Or do you still have changes to do?

**Comment by mcostalba** [comment]
It is ready.

**Comment by zamar** [comment]
approved

**Pull request merged**